### ISSUE TITLE
Consider using buf for linting and clang-format for formatting

### ISSUE THREAD

**Issue by buyology** [temporalio/temporal-proto#24]
First of all: It's so exciting to see the protos here as it means that the nice gRPC/Protobuf ecosystem can be leveraged and see Temporal expand into an array of languages!

Consider using [`buf`](https://github.com/bufbuild/buf) for [linting](https://buf.build/docs/lint-overview) against the [Protobuf style guide](https://developers.google.com/protocol-buffers/docs/style):

```sh
--- » buf check lint | wc -l
     823
--- » buf check lint 
…
common/enum.proto:30:3:Enum value name "Proto3" should be prefixed with "ENCODING_TYPE_".
common/enum.proto:30:3:Enum value name "Proto3" should be UPPER_SNAKE_CASE, such as "PROTO3".
common/enum.proto:30:3:Enum zero value name "Proto3" should be suffixed with "_UNSPECIFIED".
…
```

Also, consider using [`clang-format`](https://clang.llvm.org/docs/ClangFormat.html) for autoformatting the protos:

```
--- » clang-format -i **/*.proto
```

**Comment by alexshtin** [comment]
Thanks for reporting this. We definitely want to follow proto style guide. We are basically there with few upcoming changes:
1. Add UNSPECIFIED to all enums,
2. Convert all enum values to SCREAMING_CASE,
3. Convert field names to snake_case.

We won't prefix enum values with enum type though. Actually I recently removed this prefix because we found it very unnice. Therefore, we won't be able to use DEFAULT rule set for `buf` but will have to add some excpetions.
